langserver/build_context.go

```diff
@@ -94,7 +97,13 @@ func ContainingPackage(bctx *build.Context, filename string) (*build.Package, er
 	if PathHasPrefix(filename, bctx.GOROOT) {
 		srcDir = bctx.GOROOT // if workspace is Go stdlib
 	} else {
-		srcDir = bctx.GOPATH
+		srcDir = gopaths[0] // by default, choose first gopath
+		for _, gopath := range gopaths {
+			if strings.Contains(pkgDir, gopath) {
```

Comment:
See the `dirHasPrefix` bit at https://sourcegraph.com/github.com/golang/tools/-/blob/go/buildutil/util.go#L56:6 for the bulletproof implementation.